livekit-plugins/livekit-plugins-google/livekit/plugins/google/beta/realtime/realtime_api.py

```diff
@@ -213,280 +215,364 @@ def update_options(
         for sess in self._sessions:
             sess.update_options(voice=self._opts.voice, temperature=self._opts.temperature)
 
-    async def aclose(self) -> None: ...
+    async def aclose(self) -> None:
+        pass
 
 
 class RealtimeSession(llm.RealtimeSession):
     def __init__(self, realtime_model: RealtimeModel) -> None:
         super().__init__(realtime_model)
         self._opts = realtime_model._opts
         self._tools = llm.ToolContext.empty()
+        self._gemini_declarations: list[FunctionDeclaration] = []
         self._chat_ctx = llm.ChatContext.empty()
         self._msg_ch = utils.aio.Chan[ClientEvents]()
-        self._gemini_tools: list[Tool] = []
+        self._input_resampler: rtc.AudioResampler | None = None
+
+        # 50ms chunks
+        self._bstream = audio_utils.AudioByteStream(
+            INPUT_AUDIO_SAMPLE_RATE,
+            INPUT_AUDIO_CHANNELS,
+            samples_per_channel=INPUT_AUDIO_SAMPLE_RATE // 20,
+        )
+
         self._client = genai.Client(
             api_key=self._opts.api_key,
             vertexai=self._opts.vertexai,
             project=self._opts.project,
             location=self._opts.location,
         )
+
         self._main_atask = asyncio.create_task(self._main_task(), name="gemini-realtime-session")
 
         self._current_generation: _ResponseGeneration | None = None
-
-        self._is_interrupted = False
-        self._active_response_id = None
-        self._session = None
-        self._update_chat_ctx_lock = asyncio.Lock()
-        self._update_fnc_ctx_lock = asyncio.Lock()
+        self._active_session: genai.LiveSession | None = None
```

Comment:
```suggestion
        self._active_session: genai.LiveSession | None = None
        self._closing = False
```
signal that the session is closing